File: Esport-Frontend/src/constants/games.ts
/**
 * Дзеркало Esport-Backend/Common/Games.cs.
 *
 * Значення (ключі) — це те, що зберігає база й перевіряє валідатор.
 * Підписи — те, що бачить користувач. Змінюючи список тут, змініть
 * його і на бекенді.
 */
export const GAMES = ["CS2", "Valorant", "Dota2", "LeagueOfLegends"] as const;

export type Game = (typeof GAMES)[number];

export const GAME_LABELS: Record<Game, string> = {
  CS2: "CS2",
  Valorant: "Valorant",
  Dota2: "Dota 2",
  LeagueOfLegends: "League of Legends"
};

/** Підпис для значення з бекенда, яке може бути й невідомим (старі дані). */
export const gameLabel = (game: string): string =>
  (GAME_LABELS as Record<string, string>)[game] ?? game;

File: Esport-Backend/Common/Games.cs
namespace TForge.Common
{
    /// <summary>
    /// Канонічні дисципліни. Значення повинні збігатися з тими, що використовує
    /// фронтенд (Esport-Frontend/src/constants/games.ts) — інакше форма надішле
    /// значення, яке валідатор відхилить.
    /// </summary>
    public static class Games
    {
        public const string CS2 = "CS2";
        public const string Valorant = "Valorant";
        public const string Dota2 = "Dota2";
        public const string LeagueOfLegends = "LeagueOfLegends";

        public static readonly string[] All = { CS2, Valorant, Dota2, LeagueOfLegends };

        public static bool IsValid(string? game) => game != null && All.Contains(game);
    }
}
